### ISSUE TITLE
crash on samsung sm-t813

### ISSUE THREAD

**Issue by theromis** [godotengine/godot#76568]
### Godot version

4.0.2

### System information

Android

### Issue description

Sorry for declaring crash this way, but I can't get backtrace from this device, I've deployed game to google play store and it crashes on this device type, rest of androids works fine.

### Steps to reproduce

Just install game from app store, ui works fine and after clicking "Play" app crashes

### Minimal reproduction project

https://play.google.com/store/apps/details?id=com.lonje.jumpjump

**Comment by theromis** [comment]
This is everything what I can pull out from google play console
```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 6886 >>> com.lonje.jumpjump <<<

backtrace:
  #00  pc 0x00000000000a2c08  /system/vendor/lib64/hw/vulkan.msm8952.so (QglCommandBuffer::DrawIndexed(unsigned int, unsigned int, int, unsigned int, unsigned int)+72)
  #01  pc 0x0000000001c9c454  /data/app/com.lonje.jumpjump-1/lib/arm64/libgodot_android.so
```
![Screenshot 2023-04-28 at 10 45 27 PM](https://user-images.githubusercontent.com/3720229/235285969-32612ab7-2eec-4262-8b2b-9c776a474eb8.png)

**Comment by Calinou** [comment]
Which rendering method does the project use (Mobile or Compatibility)? Make sure the `.mobile` override for the rendering method is correctly set in the Project Settings.

**Comment by theromis** [comment]
@Calinou yes, I'm using Mobile method, because I'm using "glow" which looks bad in Compatibility method. Performance looks appropriate, but for now seems like have to switch to Compatibility.

**Comment by theromis** [comment]
@Calinou tried Compatibility mode, but "glow" seems like not works there at all. Is it possible to run on Mobile method but with OpenGL only (no vulkan)?

**Comment by Zireael07** [comment]
"Mobile" stands for "Vulkan Mobile", so no.

**Comment by theromis** [comment]
@Zireael07 is there any way to run OpenGL3 only on Godot 4? This is important for mobile gaming.

**Comment by Zireael07** [comment]
@theromis That's what Compatibility mode is for.

**Comment by theromis** [comment]
@Zireael07 I thought Compatibility is OpenGL2

**Comment by Zireael07** [comment]
No, it's OpenGL **3**. There's no OpenGL2 in Godot 4 anymore. You are probably getting confused because Godot 3.x used to have OpenGL2 as a low-spec offering, but currently there is almost no devices that don't support 3, so OpenGL 3 is now the low-spec option compared to Vulkan

**Comment by theromis** [comment]
@Zireael07 thank you for clarification, but why glow and gpuparicles not works in this mode

**Comment by Zireael07** [comment]
@theromis Because they aren't yet implemented in this mode (same goes for 3d shadows, incidentally). GLES3 was a very late addition to Godot 4 (very recently before beta)

**Comment by theromis** [comment]
@Zireael07 got it, will research how I can help with it.

**Comment by Calinou** [comment]
Glow isn't implemented in the Compatibility rendering method yet (there is no ETA for it).

**Comment by theromis** [comment]
Just sharing differences between this modes.
![Screenshot 2023-05-01 at 10 11 44 AM](https://user-images.githubusercontent.com/3720229/235495204-f26d7961-78f8-47ca-9a3a-8b80a7ec8279.png)
![Screenshot 2023-05-01 at 10 14 24 AM](https://user-images.githubusercontent.com/3720229/235495214-94f00af2-4fb5-4742-a268-191d3b8b50dc.png)
![Screenshot 2023-05-01 at 10 12 09 AM](https://user-images.githubusercontent.com/3720229/235495227-cfbf9612-cc2f-4516-85c0-5299adaddcd9.png)
![Screenshot 2023-05-01 at 10 14 54 AM](https://user-images.githubusercontent.com/3720229/235495236-5f8badab-f9bd-411d-b7fb-9a36f9cf24f0.png)
Am I right, that in 2-3 years planned to not have any differences, just a little performance drop on OpenGL3 because of Vulkan threaded execution?

**Comment by Calinou** [comment]
>  Am I right, that in 2-3 years planned to not have any differences, just a little performance drop on OpenGL3 because of Vulkan threaded execution?

The Compatibility rendering method will remain more low-end oriented. It does not aim for feature parity with Forward Mobile.

Also, Vulkan is not necessarily faster than OpenGL. In fact, it can be lot slower in certain scenes currently.

**Comment by theromis** [comment]
@Calinou this is was great explanation, thank you, I really want make godot better (at least for my game LOL) please point me where I can take glow implementation for OpenGL3 version and I'll create PR for it.

**Comment by Calinou** [comment]
>  I really want make godot better (at least for my game LOL) please point me where I can take glow implementation for Vulkan and I'll adopt it for OpenGL3.

It's more complicated than that: https://github.com/godotengine/godot/issues/66455

**Comment by theromis** [comment]
@Calinou about artifact just look at it here https://user-images.githubusercontent.com/3720229/235495227-cfbf9612-cc2f-4516-85c0-5299adaddcd9.png

**Comment by Calinou** [comment]
> @Calinou about artifact just look at it here [user-images.githubusercontent.com/3720229/235495227-cfbf9612-cc2f-4516-85c0-5299adaddcd9.png](https://user-images.githubusercontent.com/3720229/235495227-cfbf9612-cc2f-4516-85c0-5299adaddcd9.png)

Can you upload a minimal reproduction project?

**Comment by theromis** [comment]
Will create tonight, about glow seems like it about FXAA https://github.com/godotengine/godot/issues/69463#issuecomment-1338097256
can I pull it from https://github.com/naturally-intelligent/godot-4-glow-2d ?

**Comment by theromis** [comment]
@Calinou after reboot this artifact went away, seems like my local issues.

I think Godot philosophy looses one key point, in most cases mobile developers prefer loose performance for stability, and "Compatibility mode" is very important for mobile dev, I mean having same functionality as "Mobile mode" just maybe giving some warnings/notifications you are using FXAA it will significantly slow down mobile platform but give ability to avoid linking with Vulkan.

**Comment by whendricso** [comment]
Confirmed, bug and issue found on Galaxy Tab S2.

**Suggestion:** default to compatibility mode renderer after the editor crashes on Android

**Comment by whendricso** [comment]
> Just sharing differences between this modes. Compatibility: ![Screenshot 2023-05-01 at 10 12 09 AM](https://user-images.githubusercontent.com/3720229/235495227-cfbf9612-cc2f-4516-85c0-5299adaddcd9.png) Mobile: ![Screenshot 2023-05-01 at 10 14 24 AM](https://user-images.githubusercontent.com/3720229/235495214-94f00af2-4fb5-4742-a268-191d3b8b50dc.png) Compatibility: ![Screenshot 2023-05-01 at 10 11 44 AM](https://user-images.githubusercontent.com/3720229/235495204-f26d7961-78f8-47ca-9a3a-8b80a7ec8279.png) Mobile: ![Screenshot 2023-05-01 at 10 14 54 AM](https://user-images.githubusercontent.com/3720229/235495236-5f8badab-f9bd-411d-b7fb-9a36f9cf24f0.png) Am I right, that in 2-3 years planned to not have any differences, just a little performance drop on OpenGL3 because of Vulkan threaded execution?

I agree, we should aim to have no noticeable differences but actually achieving that ideal is quite an undertaking.

Let's focus on the fact that you can soft-lock your project (causing the editor to crash on startup) if you set your render mode to anything but "compatibility" on certain devices.

Therefore, I propose that we remedy the soft-lock situation first by opening the editor in compatibility mode by default if it crashed during the previous session, perhaps with a warning or popup

**Comment by Calinou** [comment]
> Therefore, I propose that we remedy the soft-lock situation first by opening the editor in compatibility mode by default if it crashed during the previous session, perhaps with a warning or popup

The issue is that we don't have a way to detect whether the app previously crashed on Android, since we currently lack a crash handler there. Implementing an Android crash handler like we have on desktop platforms isn't trivial.
